### PR TITLE
ddl, planner: handle sequence name check in preprocess

### PR DESCRIPTION
**Pull request by AilinKid**
### What problem does this PR solve?
Now preprocess phase will check sequence name if it does not exist, 
But we only know an object is not the sequence when execute evaluating it.
It's in totally different two-phase.

### What is changed and how it works?
do the both check in the same preprocessing phase.

### Check List 

Tests 

- Unit test
- Integration test

### CODE REVIEW

File: planner/core/preprocess.go [hunk 1]
```diff
@@ -232,6 +239,10 @@ func (p *preprocessor) Leave(in ast.Node) (out ast.Node, ok bool) {
 				x.Args[0] = ast.NewValueExpr(0, "", "")
 			}
 		}
+
+		if x.FnName.L == ast.NextVal || x.FnName.L == ast.LastVal || x.FnName.L == ast.SetVal {
+			p.flag &= ^inSequenceFunction
+		}
```
Comment (djshow832) [inline planner/core/preprocess.go:245]: Test this by executing a SQL with both sequence and normal table.
e.g. `select seq.nextval(), t.id from t`.

File: planner/core/preprocess.go [hunk 2]
```diff
@@ -835,8 +846,16 @@ func (p *preprocessor) handleTableName(tn *ast.TableName) {
 		p.err = err
 		return
 	}
-	tn.TableInfo = table.Meta()
+	tableInfo := table.Meta()
 	dbInfo, _ := p.is.SchemaByName(tn.Schema)
+	// tableName should be checked as sequence object.
+	if p.flag&inSequenceFunction > 0 {
+		if !table.Meta().IsSequence() {
```
Comment (djshow832) [inline planner/core/preprocess.go:853]:
```suggestion
		if !tableInfo.IsSequence() {
```

File: planner/core/preprocess.go (hunk 1 above)
Comment (AilinKid) [reply planner/core/preprocess.go:245]: addressed

File: planner/core/preprocess.go (hunk 2 above)
Comment (AilinKid) [reply planner/core/preprocess.go]: good catch

**Comment by djshow832** [comment]
I think the title or PR should be `ddl, planner: ...`, not `preprocess: ...`.

**Pull request merged**